src/main/java/com/scalefocus/blog_api/controller/BlogController.java

```diff
@@ -20,48 +23,60 @@ public class BlogController {
-        return new ResponseEntity<>(blogService.getAllBlogs(),HttpStatus.OK);
+    public ResponseEntity<List<BlogDto>> getAllBlogs() {
+        return new ResponseEntity<>(blogService.getAllBlogs(), HttpStatus.OK);
+    }
+
+    //authenticated users can get other user's posts
+    @GetMapping("/users")
+    public ResponseEntity<List<UserBlogResponse>> getUserBlogs() {
+        return new ResponseEntity<>(blogService.getUsersBlogs(), HttpStatus.OK);
     }
 
     //users can update specific blog
     @PutMapping("/{blogId}")
     public ResponseEntity<BlogDto> updateBlog(@PathVariable Long blogId,
-                                              @RequestBody BlogUpdateRequest blogUpdateRequest){
+                                              @RequestBody BlogUpdateRequest blogUpdateRequest) {
         return new ResponseEntity<>(blogService.updateBlog(blogId, blogUpdateRequest), HttpStatus.OK);
     }
 
     //users can add new tag to existing blogs
     @PutMapping("/{blogId}/tags")
-    public ResponseEntity<BlogDto> addTagToBlog(@PathVariable Long blogId, @RequestBody TagAddRequest tagAddRequest){
+    public ResponseEntity<BlogDto> addTagToBlog(@PathVariable Long blogId, @RequestBody TagAddRequest tagAddRequest) {
         return new ResponseEntity<>(blogService.addTag(blogId, tagAddRequest), HttpStatus.OK);
     }
 
     //users can remove any tag from specific blog
     @DeleteMapping("/{blogId}/tags/{tagId}")
-    public ResponseEntity<BlogDto> deleteTagFromBlog(@PathVariable Long blogId, @PathVariable Long tagId){
+    public ResponseEntity<BlogDto> deleteTagFromBlog(@PathVariable Long blogId, @PathVariable Long tagId) {
         return new ResponseEntity<>(blogService.removeTag(blogId, tagId), HttpStatus.OK);
     }
 
     //users can get all blogs by specific tagName
     @GetMapping("/tagName/{tagName}")
-    public ResponseEntity<List<BlogDto>> getAllBlogsByTagName(@PathVariable String tagName){
-        return new ResponseEntity<>(blogService.getBlogsByTagName(tagName),HttpStatus.OK);
+    public ResponseEntity<List<BlogDto>> getAllBlogsByTagName(@PathVariable String tagName) {
+        return new ResponseEntity<>(blogService.getBlogsByTagName(tagName), HttpStatus.OK);
     }
 
+    //users can get simplified blog list
     @GetMapping("/simplified")
-    public ResponseEntity<List<SimplifiedBlogResponse>> getSimplifiedBlogs(){
-        return new ResponseEntity<>(blogService.getSimplifiedBlogs(),HttpStatus.OK);
+    public ResponseEntity<List<SimplifiedBlogResponse>> getSimplifiedBlogs() {
+        return new ResponseEntity<>(blogService.getSimplifiedBlogs(), HttpStatus.OK);
     }
 
+    //authenticated users can remove own blogs
+    @DeleteMapping("/{blogId}/users/{userId}")
+    public ResponseEntity<UserResponse> deleteUserBlog(@PathVariable Long blogId, @PathVariable Long userId) {
+        return new ResponseEntity<>(blogService.deleteUserBlog(blogId, userId), HttpStatus.OK);
```

Comment:
there is no need for customer response with user Id and blog Id here. Status OK is enough

Reply:
Done.